source/utils/types.ts

```diff
@@ -67,20 +67,22 @@ export interface Response extends http.IncomingMessage {
 	request: { options: NormalizedOptions };
 }
 
-export type RetryFunction = (retry: number, error: Error | GotError | ParseError | HTTPError | MaxRedirectsError) => number;
+export type RetryFunction = (retry: number, retryOptions: NormalizedRetryOptions, error: Error | GotError | ParseError | HTTPError | MaxRedirectsError, computedValue: number) => number;
```

Comment:
I think we need a clearer name for the `retry` argument.

And would be good to add doc comments for at least `computedValue` as it's not clear to be what that is.

There's a lot of arguments now. Since it's a breaking change anyway, maybe we should just pass in one object as an argument?

Reply:
> I think we need a clearer name for the retry argument.

`iterationCount`?

> There's a lot of arguments now. Since it's a breaking change anyway, maybe we should just pass in one object as an argument?

Sounds good.

Reply:
Is it the number of retries including the original try or not? In p-retry I use `attemptNumber`, but that including the original try. https://github.com/sindresorhus/p-retry

Reply:
It is. What about `attemptCount`? Or just plural: `attempts`?

Reply:
> attemptCount

👍